src/sql/parts/modelComponents/button.component.ts

```diff
@@ -93,6 +153,14 @@ export default class ButtonComponent extends ComponentBase implements IComponent
 		this.setPropertyFromUI<sqlops.ButtonProperties, string>(this.setValueProperties, newValue);
 	}
 
+	public get iconPath(): string | URI | { light: string | URI; dark: string | URI } {
+		return this.getPropertyOrDefault<sqlops.ButtonProperties, string | URI | { light: string | URI; dark: string | URI }>((props) => props.iconPath, undefined);
+	}
+
+	public set iconPath(newValue: string | URI | { light: string | URI; dark: string | URI }) {
```

Comment:
You might want to create a type alias for this to avoid having to type it everywhere and modify it everywhere if it ever changes, even if it's not worth having an exported type for it in the sqlops API

You could do something like this above the class definition:
`type IconPath = string | URI | { light: string | URI; dark: string | URI};`